### PR TITLE
Enhancement: Use stages on Travis

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] uses build stages on Travis

Blocks #177.

💁‍♂️ For reference, see https://docs.travis-ci.com/user/build-stages/.

### CODE REVIEW

**Comment by Majkl578** [comment]
Can you please use same setup as we use in other Doctrine repos where Stages are already introduced? I don't really like having different setups for different projects... Thanks.

**Comment by localheinz** [comment]
@Majkl578 

Sure, using [`doctrine/migrations`](https://github.com/doctrine/migrations/blob/7045dd2c2e5db313de4c0434b09b59a94281254c/.travis.yml) as a basis then, although I have to admit I don't find it very readable.

**Comment by Majkl578** [comment]
> I have to admit I don't find it very readable.

The same could be said about the proposed change here - i.e. I've never seen & or* in the config. I'm not opposed to improvements, but let's keep it consistent if possible and propose changes afterwards.

**Comment by localheinz** [comment]
@Majkl578 

I can recommend http://blog.daemonl.com/2016/02/yaml.html, then!

**Comment by localheinz** [comment]
@Majkl578 

I can recommend http://blog.daemonl.com/2016/02/yaml.html, then! 🤓

File: .travis.yml [hunk 1]
```diff
@@ -1,25 +1,31 @@
...
+  - ./vendor/bin/phpunit
+
+jobs:
+  allow_failures:
+    - php: nightly
+
+  include:
+    - stage: Lint
+      before_script:
+        - travis_retry composer require --dev --prefer-dist --prefer-stable phpstan/phpstan:^0.7
```
Comment (lcobucci) [inline .travis.yml]: We need to upgrade phpstan to `^0.9`!

**Comment by lcobucci** [comment]
@localheinz 🚢 thanks!

**Comment by localheinz** [comment]
Thank you, @lcobucci and @Majkl578!

**Pull request merged**